### PR TITLE
Enforce disabled ANASAZI_TEUCHOS_TIME_MONITOR

### PR DESCRIPTION
**Pull request by masterleinad**
I just get segmentation faults when running with multiple threads if this preprocessor flag is active.

### CODE REVIEW

**Comment by masterleinad** [comment]
When changing the corresponding `CMake` option, building didn't do anything. I am not sure if we can just undefine the variable in our code safely.

File: include/mfmg/dealii/amge_host.templates.hpp [hunk 1]
```diff
@@ -215,6 +215,11 @@ void anasazi_compute_eigenvalues_and_eigenvectors(
     std::vector<std::complex<double>> &eigenvalues,
     std::vector<dealii::Vector<double>> &eigenvectors)
 {
+#ifdef ANASAZI_TEUCHOS_TIME_MONITOR
```
Comment (dalg24) [inline include/mfmg/dealii/amge_host.templates.hpp]: Would that be defined by Trilinos?

Comment (dalg24) [reply include/mfmg/dealii/amge_host.templates.hpp]: Is it something that we could/should treat at configuration time or are we just making sure we don't accidentally call that function at runtime?

Comment (masterleinad) [reply include/mfmg/dealii/amge_host.templates.hpp]: We should just configure `Trilinos` appropriately.

Comment (masterleinad) [reply include/mfmg/dealii/amge_host.templates.hpp]:
```
//Build with Teuchos TimeMonitors enabled.
Anasazi_ENABLE_TEUCHOS_TIME_MONITOR:BOOL=OFF
```

File: cmake/SetupTPLs.cmake [hunk 2]
```diff
@@ -19,6 +19,26 @@ IF(NOT DEAL_II_WITH_TRILINOS)
           "Error! deal.II must be compiled with Trilinos support.")
 ENDIF()
 
+INCLUDE(CheckCXXSourceCompiles)
+SET(CMAKE_REQUIRED_INCLUDES ${DEAL_II_INCLUDE_DIRS})
+CHECK_CXX_SOURCE_COMPILES(
```
Comment (dalg24) [inline cmake/SetupTPLs.cmake]: Trilinos include directories come from the deal.II variable?

Comment (masterleinad) [reply cmake/SetupTPLs.cmake]: Yes, we find `Trilinos` only through `deal.II`.

**Comment by masterleinad** [comment]
The updated container is at https://hub.docker.com/r/masterleinad/mfmg-stack.

**Comment by Rombur** [comment]
@masterleinad  just push it here: https://cloud.docker.com/repository/docker/rombur/mfmg-stack/general with the tag 19.06.0

**Comment by masterleinad** [comment]
I pushed it to your repository as well.

File: cmake/SetupTPLs.cmake [hunk 3]
```diff
@@ -19,6 +19,28 @@ IF(NOT DEAL_II_WITH_TRILINOS)
...
+  #ifdef ANASAZI_TEUCHOS_TIME_MONITOR
+  error
+  #endif
+  int main(){}
+  "
+  MFMG_APPROPRIATE_ANASAZI
+  )
+UNSET(CMAKE_REQUIRED_INCLUDES)
+IF(NOT MFMG_APPROPRIATE_ANASAZI)
+  MESSAGE(FATAL_ERROR
```
Comment (Rombur) [inline cmake/SetupTPLs.cmake]: Don't use `FATAL_ERROR`, we use `SEND_ERROR`

**Pull request merged**